### PR TITLE
add logging system use logrus

### PR DESCRIPTION
**Pull request by bagasteguhimani**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

no logging in backend to make sure it's easier to debug in production environment

## What is the new behavior?

Add logs using logrus

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: backend/routes.go [hunk 1]
```diff
@@ -4,12 +4,22 @@ import (
 	"backend/controllers"
 	"net/http"
 
+	"github.com/sirupsen/logrus"
+	ginlogrus "github.com/toorop/gin-logrus"
```
Comment (krismp) [inline backend/routes.go:8]: ini hapus aja @bagasteguhimani

File: backend/controllers/feedback.go [hunk 2]
```diff
@@ -49,6 +50,11 @@ func (ctrl *Controller) CreateFeedback(ctx *gin.Context) {
 
 	var lastActivity models.VisitorActivity
 	if err := ctrl.Model.GetCurrentVisitorActivity(&lastActivity, feedbackInput.SessionID, serviceId); err != nil {
+		log.WithFields(log.Fields{
+			"error": err,
+			"data":  lastActivity,
```
Comment (krismp) [inline backend/controllers/feedback.go:55]: di method `GetCurrentVisitorActivity` kan ada input `feedbackInput.SessionID` sama `serviceId` juga @bagasteguhimani , nah kaya info kaya gitu berguna waktu debugging, jadi dimasukan aja ke log:
```go
log.WithFields(log.Fields{
			"error": err,
			"data":  lastActivity,
			"session_id": feedbackInput.SessionID,
			"service_id": serviceId,
}).Error("Error on get current visitor activity!")
```

Comment (krismp) [reply backend/controllers/feedback.go:55]: Coba cek log yg lain ya @bagasteguhimani , sertakan input yg belum masuk ke log, kecuali context gak perlu di masukan

File: backend/controllers/request.go [hunk 3]
```diff
@@ -28,6 +28,11 @@ func RequestToServiceAnalytics(ctx *gin.Context, service models.Service, inputDa
 	serviceData, err = RequestToAnalyticSync(dataAnalytic, service.Slug)
 
 	if err != nil {
+		log.WithFields(log.Fields{
+			"error": err,
+			"data":  serviceData,
```
Comment (krismp) [inline backend/controllers/request.go:33]: err disini kan return dari method `serviceData, err = RequestToAnalyticSync(dataAnalytic, service.Slug)` ya ?
nah disitu yg dimasukin input nya, `dataAnalytic` sama `service.Slug` harusnya @bagasteguhimani 

coba cari case serupa dan di update ya

File: backend/controllers/service.go [hunk 4]
```diff
@@ -19,6 +19,10 @@ func (ctrl *Controller) GetServices(ctx *gin.Context) {
 	// and it has an invalid value
 	isValid, serviceType := ctrl.Model.IsValidServiceType(serviceTypeQuery)
 	if isAnyQueryType && !isValid {
+		log.WithFields(log.Fields{
+			"data": serviceTypeQuery,
+		}).Error("Error on get services!")
```
Comment (krismp) [inline backend/controllers/service.go:24]: message nya juga di sesuaikan sm yg di lakukan di atasnya, di sini `Error on get services` tp sebenernya error karena query params yg di pass oleh user gak valid ya? nah coba di ganti message nya yg cukup representatif

File: backend/controllers/innovation.go [hunk 5]
```diff
@@ -20,25 +22,40 @@ func RequestToInnovationSync(postBody []byte, innovationSlug string) (models.Ser
...
 	if err != nil {
-		log.Fatal(err)
+		log.WithFields(log.Fields{
+			"error":    err,
+			"data":     data,
+			"payload":  payload,
+			"BASE_URL": BASE_URL,
+			"slug":     innovationSlug,
+			"Method":   "POST",
+		}).Fatal("Fatal on send http new request!")
```
Comment (krismp) [inline backend/controllers/innovation.go:32]: ini jgn fatal deh, error aja @bagasteguhimani

File: backend/controllers/session.go [hunk 6]
```diff
@@ -2,16 +2,23 @@ package controllers
...
 
 func (ctrl *Controller) IsSessionExist(sessionId string) bool {
 	var visitor models.Visitor
 
 	if err := ctrl.Model.GetVisitor(&visitor, sessionId); err != nil {
+
+		log.WithFields(log.Fields{
+			"error":      err,
+			"session_id": sessionId,
+		}).Fatal("Fatal, Session Id not Exist")
```
Comment (krismp) [inline backend/controllers/session.go:20]: yg Fatal semua ganti Error aja dulu ya @bagasteguhimani

**Review by krismp — COMMENTED** [review]
all goods abis comment ini

File: backend/main.go [hunk 7]
```diff
@@ -7,11 +7,22 @@ import (
...
+
+	// Log as JSON instead of the default ASCII formatter.
+	log.SetFormatter(&log.JSONFormatter{})
+
+	// Output to stdout instead of the default stderr
+	// Can be any io.Writer, see below for File example
+	log.SetOutput(os.Stdout)
+
+	// Only log the warning severity or above.
+	log.SetLevel(log.WarnLevel)
```
Comment (krismp) [inline backend/main.go:25]: log severity nya masukin ,e .env aja @bagasteguhimani LOG_LEVEL, soal env bisa tanya @sonadztux atau @lazuardyk

File: backend/models/face_payment.go [hunk 8]
```diff
@@ -103,6 +111,12 @@ func (m *Model) CreateAccountWallet(sessionId string, newAccountWallet *FacePaym
 
 	err = m.DBConn.Where("session_id = ? AND is_active = ?", sessionId, "true").Last(&newAccount).Error
 	if err != nil {
+		log.WithFields(log.Fields{
+			"error":      err,
+			"data":       newAccountWallet,
+			"session_id": sessionId,
+			"is_active":  newAccount.IsActive,
+		}).Error("Error on create account wallet to DB!")
```
Comment (krismp) [inline backend/models/face_payment.go:119]: error nya waktu dapetin last account dengan session id

**Review by krismp — COMMENTED** [review]
@bagasteguhimani error di model ada yg belum di kasih log, sama coba cek lagi message nya udh relevan blm 

controllers all good, tinggal model nya, please check ya, thx

**Comment by sonadztux** [comment]
Label PR sama changelog jangan lupa nanti ditambahin.

**Comment by sonadztux** [comment]
Sama link ke issue-nya.

**Review by krismp — APPROVED** [review]

**Review by sonadztux — APPROVED** [review]
ok lgtm, sisa changelog jangan lupa

**Pull request merged**